tephra-core/src/main/java/co/cask/tephra/TransactionManager.java

```diff
@@ -905,6 +906,11 @@ private void doCommit(long writePointer, Set<ChangeId> changes, long commitPoint
     }
     // moving read pointer
     moveReadPointerIfNeeded(writePointer);
+    if (checkpointWritePointers != null) {
+      for (long checkpointWritePointer : checkpointWritePointers) {
```

Comment:
Filed JIRA https://issues.cask.co/browse/TEPHRA-141 to reconcile usages of transaction ids v/s write pointers.